lib/js/jslib.js

```diff
@@ -1757,11 +1757,30 @@ function _getInputValue(id) {
   var element = document.getElementById(id);
   DEBUG.assert(element != null, "invalid input node (id " + id + ")");
   DEBUG.assert(element.value != undefined, "invalid input value in id " + id)
-  return element.value;
+  if ((element.type != "radio" && element.type != "checkbox") || element.checked) {
+    return element.value;
+  } else {
+    return "";
+  }
 }
 
 var getInputValue = LINKS.kify(_getInputValue);
 
+// getRadioGroupValue : [String] -> String
+function _getRadioGroupValue(ids) {
+  let ptr = ids;
+  while ( LINKEDLIST.isNil(ptr) == false ) {
+    var element = document.getElementById(ptr._head);
+    if (element.checked) {
+      return element.value;
+    }
+    ptr = ptr._tail;
+  }
+  return "";
```

Comment:
Comment: we ought to define `String.empty` for the empty string somewhere in jslib.

Reply:
There are around 20 occurrences of `""` in `jslib.js`.  Should they all be replaced by `String.empty` or just some of them?

Reply:
Quite possibly! I am not familiar with every instance of `""` in `jslib.js`. My thinking is that we ought to have a module for each primitive type to ensure some consistency. But I think this would be implemented in a separate patch.